arangod/VocBase/vocbase.cpp

```diff
@@ -1835,6 +1967,43 @@ bool TRI_vocbase_t::visitDataSources(dataSourceVisitor const& visitor, bool lock
   return true;
 }
 
+auto TRI_vocbase_t::getReplicatedLogById(arangodb::replication2::LogId id) const
+    -> std::shared_ptr<arangodb::replication2::replicated_log::ReplicatedLog> {
+  std::unique_lock guard(_logManager->_mutex);
+  return _logManager->getReplicatedLogById(id);
```

Comment:
Just wondering why the lock is acquired from the outside here, instead of doing the locking inside `_logManager`.

Reply:
Fixed